src/k2/cmd/ycsb/data.h

```diff
@@ -0,0 +1,155 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
+
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in all copies
+    or
+    substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS",
+    WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+    DAMAGES OR OTHER
+    LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+#pragma once
+
+#include <decimal/decimal>
+#include <string>
+
+#include <k2/common/Common.h>
+#include <k2/module/k23si/client/k23si_client.h>
+#include <k2/transport/Payload.h>
+#include <k2/transport/PayloadSerialization.h>
+
+#include "ycsb_rand.h"
+#include "Log.h"
+using namespace k2;
+static const String ycsbCollectionName = "YCSB";
+
+// helper function to convert to base (default 93 (ascii from 33 to 126)) representation
+String itostring(int n, int base = 93)
+{
+    String buf(String::initialized_later{}, 1000);
```

Comment:
There is a possibility of buffer overflow doing it this way. Is it possible to get the needed length arithmetically here? If not, keep track of how many characters you use and call "append" once you have used up the initial allocation. Then I would also start with a smaller size, maybe 100.